src/Axe/Controllers/CompilerController.cs

```diff
@@ -0,0 +1,97 @@
+
+        public CompilerController(UserManager<ApplicationUser> userManager, AxeDbContext context)
+            : base(userManager, context)
+        {
+            this._context = context;
+        }
+
+        [HttpGet]
+        public IActionResult Index()
+        {
+            List<CodeBlock> codeBlocks = this._context.CodeBlock.ToList();
+            return View(codeBlocks);
+        }
+
+        [HttpPost]
+        public IActionResult Execute(CodeBlock codeBlock)
+        {
+            if (ModelState.IsValid)
+            {
+                if (codeBlock != null)
+                {
+                    string assemblyName = "AxeCompilerAssembly" + Path.GetRandomFileName();
+                    // Random file name is necessary because .NETCore doesn't allow to unload assembly
+                    SyntaxTree syntaxTree = CSharpSyntaxTree.ParseText(codeBlock.Code);
+
+                    MetadataReference[] references = new MetadataReference[]
+                    {
+                    MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location),
+                    MetadataReference.CreateFromFile(typeof(Hashtable).GetTypeInfo().Assembly.Location)
+                        // MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
+                        // MetadataReference.CreateFromFile(typeof(Enumerable).Assembly.Location)
+                    };
+
+                    CSharpCompilation compilation = CSharpCompilation.Create(
+                        assemblyName: assemblyName,
+                        syntaxTrees: new[] { syntaxTree },
+                        references: references,
+                        options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+                    using (var stream = new MemoryStream())
+                    {
+                        EmitResult result = compilation.Emit(stream);
+                        if (!result.Success)
+                        {
+                            IEnumerable<Diagnostic> failures = result.Diagnostics.Where(diagnostic =>
+                                diagnostic.IsWarningAsError ||
+                                diagnostic.Severity == DiagnosticSeverity.Error);
+
+                            foreach (Diagnostic diagnostic in failures)
+                            {
+                                Console.Error.WriteLine("{0}: {1}", diagnostic.Id, diagnostic.GetMessage());
+                            }
+                            ViewData["StatusOfExecuting"] = "Error";
+                        }
+                        else
+                        {
+                            stream.Seek(0, SeekOrigin.Begin);
+
+                            AssemblyLoadContext context = AssemblyLoadContext.Default;
+                            Assembly assembly = context.LoadFromStream(stream);
+                            MethodInfo method = assembly.GetType("RoslynCompileSample.Writer").GetMethod("Write");
```

Comment:
needs discussion - why `Writer` and `Write` are hardcoded here?

Reply:
I use Roslyn .NET platform for compiling and executing code. Roslyn requires such a format that something to execute it's necessary to get it. It isn't problem because name of namespace, class and method is written in task.